source/interact-data/modify-results.txt

```diff
@@ -0,0 +1,281 @@
+   you cannot specify an association and its fields in a projection in
+   the same query.
+
+If a document has ``has_one`` or ``has_and_belongs_to_many``
+associations, you must include the fields with foreign keys in the list
+of attributes loaded when using ``only()`` for those associations to be
+loaded.
+
+In the following example, the ``Band`` and ``Manager`` models have a
+``has_and_belongs_to_many`` association:
+
+.. code-block:: ruby
+
+   class Band
+     include Mongoid::Document
+     field :name, type: String
+     has_and_belongs_to_many :managers
+   end
+
+   class Manager
+     include Mongoid::Document
+     has_and_belongs_to_many :bands
+   end
+
+The following code demonstrates how {+odm+} can load the associated
+``Manager`` objects if you include the ``manager_ids`` field:
+
+.. literalinclude:: /includes/interact-data/modify-results.rb
+   :start-after: start-only-embed-association
+   :end-before: end-only-embed-association
+   :language: ruby
+   :dedent:
+
+Exclude Fields
+~~~~~~~~~~~~~~
+
+You can explicitly exclude fields from results by using the
+``without()`` method.
+
+The following code excludes the ``year`` field from returned ``Band``
+objects:
+
+.. literalinclude:: /includes/interact-data/modify-results.rb
+   :start-after: start-without
+   :end-before: end-without
+   :language: ruby
+   :dedent:
+
+.. important:: _id Field
+
+   {+odm+} requires the ``_id`` field for various operations, so you
+   *cannot* exclude the ``_id`` field or the ``id`` alias from results.
+   If you pass ``_id`` or ``id`` to the ``without()`` method, {+odm+}
+   ignores it.
+
+.. _mongoid-data-sort:
+
+Sort Results
+------------
+
+You can sort the order that {+odm+} returns documents by using the
+``order()`` and ``order_by()`` methods.
+
+These methods accept a hash that indicates which fields to order the
+documents by, and whether to use an ascending or descending order for
+each field.
+
+You can specify the sort direction in the following ways:
+
+- Integers ``1`` (ascending) and ``-1`` (descending)
+  - Example: ``Band.order(name: 1, year: -1)``
+
+- Symbols ``:asc`` and ``:desc``
+  - Example: ``Band.order(name: :asc)``
+
+- Strings ``"asc"`` and ``"desc"``
+  - Example: ``Band.order_by(name: "asc", year: "desc")``
+
+The ``order()`` method also accepts the following sort specifications:
+
+- Array of two-element arrays:
+
+  - Strings
+    - Example: ``Band.order([['name', 'asc'], ['year', 'desc']])``
+
+  - Symbols
+    - Example: ``Band.order([[:name, :asc]])``
+
+- ``asc`` and ``desc`` methods on symbols
+  - Example: ``Band.order(:name.asc, :year.desc)``
+
+- SQL syntax
+  - Example: ``Band.order('name desc')``
+
+You can also use the ``asc()`` and ``desc()`` methods instead of using
+``order()``:
+
+.. code-block:: ruby
+
+   Band.asc('name').desc('year')
+
+If you chain sort specifications, the first call defines the most
+significant criteria and the newest call defines the least significant
+one.
```

Comment:
Q: does most/least significant mean the order in which ruby will try to sort the fields by (sort the most significant, then second most, etc.)? I just haven't heard these terms before.

Reply:
Ill reword